### DISCUSSION TITLE
Application Crash Issue with Data Binding in .NET MAUI 8 CollectionView and ObservableCollection

### DISCUSSION THREAD

**Discussion by gauravK996** [dotnet/maui#20040] [General]
We are currently facing an issue related to data binding when using .NET MAUI's CollectionView in conjunction with ObservableCollection. The problem appears to be affecting the expected behavior of the data binding mechanism, resulting in unexpected outcomes. Application consistently crashes, specifically when multiple data items are added in bulk and scrolling is performed within the CollectionView.
Temporary workaround: It appears that introducing a Task.Delay(50) in specific scenarios helps mitigate the problem.
But the application continues to crash despite the introduction of Task.Delay

**Comment by pmichale** [comment]
Have you ever found a solution for this issue?

**Reply by bronteq** [reply]
Look at the open issue, I found a work around: https://github.com/dotnet/maui/issues/20037
Please vote the issue with thumbs up emoji to put more priority, thanks

**Comment by Bnjbel97** [comment]
I have seen similar behavior.  I can tell you some workarounds that worked for me.  

- I had a collection view that was being updated constantly - server communication and gps data(different collection).  The lag that was most notable was dumping the first 150 messages once at 300 messages. I do something similar to the below link.  
 
https://dev.to/vhugogarcia/observablerangecollection-in-net-maui-k9j

- Another issue was sorting large data collections that are constantly being updated through server communications.  In WPF, there was a better way of tying in sorts to observable collections however I have not found an alternative yet.  In this particular case I referenced a different observable collection that is bound to the UI.  The data coming in is a simple list (dictionary would probably be more efficient). Data coming in does its processing, etc. The UI Observable collection handles sorting, logic from the non ui collection -> set UI Collection. Is via reference, so as data changes within the list so does the ui collection.
